android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java

```diff
@@ -362,70 +362,97 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    .setBoolean(Pref.HTTPS_ONLY_MODE_ENABLED, (boolean) newValue);
         } else if (PREF_DE_AMP.equals(key)) {
             UserPrefs.get(Profile.getLastUsedRegularProfile())
                     .setBoolean(BravePref.DE_AMP_PREF_ENABLED, (boolean) newValue);
         } else if (PREF_DEBOUNCE.equals(key)) {
             UserPrefs.get(Profile.getLastUsedRegularProfile())
                     .setBoolean(BravePref.DEBOUNCE_ENABLED, (boolean) newValue);
         } else if (PREF_IPFS_GATEWAY.equals(key)) {
-            BravePrefServiceBridge.getInstance().setIpfsGatewayEnabled((boolean) newValue);
+            if (BravePrivacySettingsIPFSUtils.isIPFSSupported()) {
+                @IPFSResolveMethodTypes
+                final int value = (boolean) newValue ? IPFSResolveMethodTypes.IPFS_ASK
+                                                     : IPFSResolveMethodTypes.IPFS_DISABLED;
+                UserPrefs.get(Profile.getLastUsedRegularProfile())
+                        .setInteger(BravePref.IPFS_RESOLVE_METHOD, value);
+            }
         } else if (PREF_BLOCK_COOKIE_CONSENT_NOTICES.equals(key)) {
             if (mCookieListOptInPageAndroidHandler != null) {
                 mCookieListOptInPageAndroidHandler.enableFilter((boolean) newValue);
             }
-
         } else if (PREF_FINGERPRINTING_PROTECTION.equals(key)) {
-            if (newValue instanceof String
-                    && String.valueOf(newValue).equals(
-                            BraveShieldsContentSettings.BLOCK_RESOURCE)) {
-                BravePrefServiceBridge.getInstance().setFingerprintingControlType(
-                        BraveShieldsContentSettings.BLOCK_RESOURCE);
-                mFingerprintingProtectionPref.setSummary(getActivity().getResources().getString(
-                        R.string.block_fingerprinting_option_1));
-                mFingerprintingProtectionPref.setCheckedIndex(0);
-            } else if (newValue instanceof String
-                    && String.valueOf(newValue).equals(BraveShieldsContentSettings.DEFAULT)) {
-                BravePrefServiceBridge.getInstance().setFingerprintingControlType(
-                        BraveShieldsContentSettings.DEFAULT);
-                mFingerprintingProtectionPref.setSummary(getActivity().getResources().getString(
-                        R.string.block_fingerprinting_option_2));
-                mFingerprintingProtectionPref.setCheckedIndex(1);
-            } else {
-                BravePrefServiceBridge.getInstance().setFingerprintingControlType(
-                        BraveShieldsContentSettings.ALLOW_RESOURCE);
-                mFingerprintingProtectionPref.setSummary(getActivity().getResources().getString(
-                        R.string.block_fingerprinting_option_3));
-                mFingerprintingProtectionPref.setCheckedIndex(2);
+            if (newValue instanceof String) {
+                switch (String.valueOf(newValue)) {
```

Comment:
we can do `String.valueOf(newValue)` conversion once as for each case we do it at least once and for two first cases we convert twice.

Reply:
I guess the compiler can handle it, but I added a line to convert the string.

Reply:
yeah, it could be